pkg/revisions/default_watcher.go

```diff
@@ -159,20 +100,19 @@ func (p *defaultWatcher) notifyHandlers() {
 	}
 }
 
-func getDefault(meta metav1.Object) string {
-	if revision, ok := meta.GetLabels()[label.IoIstioRev.Name]; ok {
-		return revision
+func (p *defaultWatcher) setDefault(key types.NamespacedName) error {
+	revision := ""
+	wh, _, _ := p.webhookInformer.GetIndexer().GetByKey(key.Name)
+	if wh != nil {
+		revision = wh.(metav1.Object).GetLabels()[label.IoIstioRev.Name]
 	}
-	return ""
-}
-
-func (p *defaultWatcher) setDefault(revision string) {
 	p.mu.Lock()
 	defer p.mu.Unlock()
 	p.defaultRevision = revision
 	p.notifyHandlers()
+	return nil
 }
 
-func filterUpdate(obj metav1.Object) bool {
-	return obj.GetName() != defaultTagWebhookName
+func filterUpdate(obj controllers.Object) bool {
```

Comment:
very nit: counterintuitive naming, filterUpdate should be true when the object is not the default tag

Reply:
I would say `filter` is a fairly well known pattern in functional programing https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#filter-java.util.function.Predicate- 

But I will admit I confused myself when writing the code though so I mostly agree